executor/batch.rs

```diff
@@ -203,6 +204,16 @@ impl Batch {
         self.entries += 1;
     }
 
+    pub(crate) fn append_mapped(
+        &mut self,
+        row: MaybeOwnedRow<'_>,
+        mapping: impl Iterator<Item = (VariablePosition, VariablePosition)>,
+    ) {
+        let mut destination_row = self.row_internal_mut(self.entries);
+        destination_row.copy_mapped(row, mapping);
+        self.entries += 1;
+    }
+
```

Comment:
Don't you need to append `1` to the multiplicities too?

Reply:
In both Append methods? Also, when you create variations in the APIs you'd have one canonical one and redirect the other one to the canonical one (append() should call append_mapped with a identity mapping)

Reply:
> Don't you need to append 1 to the multiplicities too?
I haven't applied the [append changes](https://github.com/krishnangovindraj/typedb/pull/7) to this branch.

> append() should call append_mapped with a identity mapping
That makes a lot of sense and works quite nicely, but I think `copy_from_row` might have scope for compiler optimisation, whereas `copy_mapped` definitely won't. I can still do it, but there's the general question of when it's worth doing these things